src/main/java/io/jenkins/plugins/tuleap_oauth/TuleapSecurityRealm.java

```diff
@@ -176,27 +184,56 @@ private void setClientSecret(String secretString) {
 
     @Override
     public UserDetails loadUserByUsername(String username) {
-        Authentication authenticatedUserAcegiToken = SecurityContextHolder.getContext().getAuthentication();
+        this.injectInstances();
+
+        final Authentication authenticatedUserAcegiToken = this.pluginHelper.getCurrentUserAuthenticationToken();
 
         if (authenticatedUserAcegiToken == null) {
-            throw new UsernameNotFoundException("No access token found for user " + username);
+            throw new UserMayOrMayNotExistException("No access token found for user " + username);
         }
 
         if (!(authenticatedUserAcegiToken instanceof TuleapAuthenticationToken)) {
             throw new UserMayOrMayNotExistException("Unknown token type for user " + username);
         }
 
-        AccessTokenRepresentation accessToken = this.gson.fromJson(
-            this.tuleapAccessTokenStorage.retrieve(Objects.requireNonNull(User.getById(username, false))).getPlainText(),
-            AccessTokenRepresentation.class
-        );
+        final User user = this.pluginHelper.getUser(username);
 
-        UserInfo userInfo = this.openIDClientApi.getUserInfo(accessToken);
+        if (!(user != null && this.tuleapUserPropertyStorage.has(user))) {
+            throw new UsernameNotFoundException("Could not find user " + username + " for Tuleap");
+        }
 
-        return new TuleapUserDetails(
-            userInfo.getUsername(),
-            authenticatedUserAcegiToken.getAuthorities()
-        );
+        return new TuleapUserDetails(username);
+    }
+
+    @Override
+    public GroupDetails loadGroupByGroupname(String groupName) {
+        final Authentication authenticatedUserAcegiToken = this.pluginHelper.getCurrentUserAuthenticationToken();
+
+        if (authenticatedUserAcegiToken == null) {
+            throw new UserMayOrMayNotExistException("No access token found for user");
+        }
+
+        if (!(authenticatedUserAcegiToken instanceof TuleapAuthenticationToken)) {
+            throw new UserMayOrMayNotExistException("Unknown token type for user");
+        }
+
+        final TuleapAuthenticationToken tuleapAuthenticationToken = (TuleapAuthenticationToken) authenticatedUserAcegiToken;
+
+        if (this.tokenHasTuleapGroup(tuleapAuthenticationToken, groupName)) {
+            return new TuleapGroupDetails(groupName);
+        }
+
+        throw new UsernameNotFoundException("Group " + groupName + " not found for current Tuleap user");
+    }
+
+    private Boolean tokenHasTuleapGroup(TuleapAuthenticationToken token, String groupName) {
```

Comment:
what do you think about hasTuleapGroupToken instead?

Reply:
If your not against keeping it, I would rather not to. I find it very much easier to read the sentence "if token has tuleap group" than "if has tuleap group token"